cishouseholds/derive.py

```diff
@@ -39,18 +39,18 @@ def assign_datetime_from_coalesced_columns_and_log_source(
         if type != "timestamp":
             df = df.withColumn(col, F.to_timestamp(col, format=f"{date_format} {time_format}"))
     for col in ordered_columns:
-        check_distinct = df.agg(F.countDistinct(F.date_format(col, time_format))).collect()[0][0] == 1
+        check_distinct = df.agg(F.countDistinct(F.date_format(col, "HH:mm:ss"))).collect()[0][0] == 1
         col_object = F.when(
-            (F.col(col) >= F.lit(min_date)) & (F.col(col) < F.col(file_date_column)),
+            F.col(col).between(F.lit(min_date), F.col(file_date_column)),
             F.when(
-                (F.date_format(col, time_format) == "00:00:00") & F.lit(check_distinct),
-                F.concat_ws(" ", F.date_format(col, date_format), F.lit(default_timestamp)),
+                (F.date_format(col, "HH:mm:ss") == "00:00:00") & F.lit(check_distinct),
```

Comment:
If a datetime field has it's time part set to midnight, it will be treated as a date here. I think we should take the dates as a separate list to make this simpler

Reply:
Yes please, as this assumes that our datetime fields all do contain a time part (they may not)